File: LeetCode_ProblemSet/Go/Problem624.go
package main

import "math"

func maxDistance(arrays [][]int) int {
	smallest, largest := arrays[0][0], arrays[0][len(arrays[0]) - 1]
	maxDist := 0
	for idx := 1; idx < len(arrays); idx++ {
		maxDist = max(
			int(math.Abs(float64(arrays[idx][len(arrays[idx]) - 1] - smallest))),
			int(math.Abs(float64(largest - arrays[idx][0]))),
			maxDist,
		)
		smallest = min(smallest, arrays[idx][0])
		largest = max(largest, arrays[idx][len(arrays[idx]) - 1])
	}
	return maxDist
}

File: LeetCode_ProblemSet/Problem624.py
class Solution(object):
    def maxDistance(self, arrays):
        """
        :type arrays: List[List[int]]
        :rtype: int
        """
        smallest, largest = arrays[0][0], arrays[0][-1]
        max_dist = 0
        for idx in range(1, len(arrays)):
            max_dist = max(
                abs(arrays[idx][-1] - smallest), 
                abs(largest - arrays[idx][0]),
                max_dist
            )
            smallest = min(smallest, arrays[idx][0])
            largest = max(largest, arrays[idx][-1])
        return max_dist